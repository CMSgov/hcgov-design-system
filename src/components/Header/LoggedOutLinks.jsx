import PropTypes from 'prop-types';
import React from 'react';
import defaultMenuLinks from './defaultMenuLinks';

/**
 * The list of links displayed horizontally in the header in the logged-out
 * menu. This by default is the login link and the language link.
 */
const LoggedOutLinks = ({
  locale,
  deConsumer,
  subpath,
  primaryDomain,
  switchLocaleLink,
}) => {
  const links = defaultMenuLinks(
    locale,
    deConsumer,
    subpath,
    primaryDomain,
    switchLocaleLink
  ).minimal;
  return (
    <ul className="hc-c-logged-out-links ds-c-list--bare ds-u-display--none ds-u-sm-display--inline-block">
      {links.map((link) => {
        return (
          <li key={link.href} className="hc-c-logged-out-links__li">
            <a href={link.href} className="hc-c-logged-out-links__link">
              {link.label}
            </a>
          </li>
        );
      })}
    </ul>
  );
};

LoggedOutLinks.propTypes = {
  locale: PropTypes.string.isRequired,
  /**
   * Indicates when a consumer is coming from a Direct Enrollment flow.
   * This will include additional messaging and modify some of the links.
   */
  deConsumer: PropTypes.bool,
  /**
   * [See `Header.props.subpath`]({{root}}/patterns/header/#patterns.header.react)
   */
  subpath: PropTypes.string,
};

export default LoggedOutLinks;
