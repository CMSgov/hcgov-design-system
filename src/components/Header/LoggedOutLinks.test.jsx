import LoggedOutLinks from './LoggedOutLinks';
import React from 'react';
import { shallow } from 'enzyme';

describe('LoggedOutLinks', function () {
  it('renders logged out links', () => {
    expect(shallow(<LoggedOutLinks locale="en" />)).toMatchSnapshot();
  });

  it('renders logged out links for Direct Enrollment', () => {
    expect(shallow(<LoggedOutLinks locale="en" deConsumer />)).toMatchSnapshot();
  });

  it('renders logged out links with absolute URLs', () => {
    expect(
      shallow(<LoggedOutLinks locale="en" primaryDomain="https://www.healthcare.gov" />)
    ).toMatchSnapshot();
  });

  it('renders custom locale switch link', () => {
    expect(
      shallow(
        <LoggedOutLinks locale="en" switchLocaleLink="https://ayudalocal.cuidadodesalud.gov/es" />
      )
    ).toMatchSnapshot();
  });

  it('does not render the login link if the flag hideLoginLink is set to true', () => {
    expect(shallow(<LoggedOutLinks locale="en" hideLoginLink />)).toMatchSnapshot();
  });

  it('does not render the language switch if hideLanguageSwitch is set to true', () => {
    expect(shallow(<LoggedOutLinks locale="en" hideLanguageSwitch />)).toMatchSnapshot();
  });
});
