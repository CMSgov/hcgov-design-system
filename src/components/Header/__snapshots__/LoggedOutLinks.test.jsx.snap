// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedOutLinks does not render the language switch if hideLanguageSwitch is set to true 1`] = `
<ul
  className="hc-c-logged-out-links ds-c-list--bare ds-u-display--none ds-u-sm-display--inline-block"
>
  <li
    className="hc-c-logged-out-links__li"
    key="/login"
  >
    <a
      className="hc-c-logged-out-links__link"
      href="/login"
    >
      header.login
    </a>
  </li>
</ul>
`;

exports[`LoggedOutLinks does not render the login link if the flag hideLoginLink is set to true 1`] = `
<ul
  className="hc-c-logged-out-links ds-c-list--bare ds-u-display--none ds-u-sm-display--inline-block"
>
  <li
    className="hc-c-logged-out-links__li"
    key="https://www.cuidadodesalud.gov/es/"
  >
    <a
      className="hc-c-logged-out-links__link"
      href="https://www.cuidadodesalud.gov/es/"
    >
      header.español
    </a>
  </li>
</ul>
`;

exports[`LoggedOutLinks renders custom locale switch link 1`] = `
<ul
  className="hc-c-logged-out-links ds-c-list--bare ds-u-display--none ds-u-sm-display--inline-block"
>
  <li
    className="hc-c-logged-out-links__li"
    key="/login"
  >
    <a
      className="hc-c-logged-out-links__link"
      href="/login"
    >
      header.login
    </a>
  </li>
  <li
    className="hc-c-logged-out-links__li"
    key="https://ayudalocal.cuidadodesalud.gov/es"
  >
    <a
      className="hc-c-logged-out-links__link"
      href="https://ayudalocal.cuidadodesalud.gov/es"
    >
      header.español
    </a>
  </li>
</ul>
`;

exports[`LoggedOutLinks renders logged out links 1`] = `
<ul
  className="hc-c-logged-out-links ds-c-list--bare ds-u-display--none ds-u-sm-display--inline-block"
>
  <li
    className="hc-c-logged-out-links__li"
    key="/login"
  >
    <a
      className="hc-c-logged-out-links__link"
      href="/login"
    >
      header.login
    </a>
  </li>
  <li
    className="hc-c-logged-out-links__li"
    key="https://www.cuidadodesalud.gov/es/"
  >
    <a
      className="hc-c-logged-out-links__link"
      href="https://www.cuidadodesalud.gov/es/"
    >
      header.español
    </a>
  </li>
</ul>
`;

exports[`LoggedOutLinks renders logged out links for Direct Enrollment 1`] = `
<ul
  className="hc-c-logged-out-links ds-c-list--bare ds-u-display--none ds-u-sm-display--inline-block"
>
  <li
    className="hc-c-logged-out-links__li"
    key="/login?check_de=1"
  >
    <a
      className="hc-c-logged-out-links__link"
      href="/login?check_de=1"
    >
      header.login
    </a>
  </li>
  <li
    className="hc-c-logged-out-links__li"
    key="https://www.cuidadodesalud.gov/es/"
  >
    <a
      className="hc-c-logged-out-links__link"
      href="https://www.cuidadodesalud.gov/es/"
    >
      header.español
    </a>
  </li>
</ul>
`;

exports[`LoggedOutLinks renders logged out links with absolute URLs 1`] = `
<ul
  className="hc-c-logged-out-links ds-c-list--bare ds-u-display--none ds-u-sm-display--inline-block"
>
  <li
    className="hc-c-logged-out-links__li"
    key="https://www.healthcare.gov/login"
  >
    <a
      className="hc-c-logged-out-links__link"
      href="https://www.healthcare.gov/login"
    >
      header.login
    </a>
  </li>
  <li
    className="hc-c-logged-out-links__li"
    key="https://www.cuidadodesalud.gov/es/"
  >
    <a
      className="hc-c-logged-out-links__link"
      href="https://www.cuidadodesalud.gov/es/"
    >
      header.español
    </a>
  </li>
</ul>
`;
